crates/router/src/types/storage/payment_intent.rs

```diff
@@ -85,4 +91,34 @@ impl PaymentIntentDbExt for PaymentIntent {
             .change_context(errors::DatabaseError::NotFound)
             .attach_printable_lazy(|| "Error filtering records by predicate")
     }
+
+    #[instrument(skip(conn))]
+    async fn filter_by_time_constraints(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+        time_range: &api::TimeRange,
+    ) -> CustomResult<Vec<Self>, errors::DatabaseError> {
+        let start_time = time_range.start_time;
+        let end_time = time_range
+            .end_time
+            .unwrap_or_else(common_utils::date_time::now);
+
+        //[#350]: Replace this with Boxable Expression and pass it into generic filter
+        // when https://github.com/rust-lang/rust/issues/52662 becomes stable
+        let mut filter = <Self as HasTable>::table()
+            .filter(dsl::merchant_id.eq(merchant_id.to_owned()))
+            .order(dsl::modified_at.desc())
+            .into_boxed();
+
+        filter = filter.filter(dsl::created_at.ge(start_time));
+
+        filter = filter.filter(dsl::created_at.le(end_time));
+
+        filter
+            .get_results_async(conn)
+            .await
+            .into_report()
+            .change_context(errors::DatabaseError::NotFound)
```

Comment:
We cannot directly change the context to `NotFound` for all the errors. you can remove this change context